__init__.py

```diff
@@ -30,10 +30,10 @@ def __init__(self, view, function):
 
         self.endianness = 'big' if self._view.endianness else 'little'
 
-        self.cmdline = ['retdec-decompiler.sh']
         self.cmdline.append('-m')
-        self.cmdline.append('raw')
+        self.cmdline.append(mode)
 
+        self.cmdline = ['retdec-decompiler.sh']
```

Comment:
You must init cmdline before appending `'-m'` and `mode`. Better remove your last commit.